### PR TITLE
Update workload.Uid

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
**Please provide a description of this PR:**

Respect format <cluster>/<group>/<kind>/<namespace>/<name></section-name>, remove version

resource Kind should be Upper cased

### CODE REVIEW

**Review by howardjohn — CHANGES_REQUESTED** [review]
Can you provide context behind this change? I don't think its correct. Version is not part of a unique identifier in Kubernetes -- If I create a VirtualService, I get a v1alpha3 and v1beta1 that are _the same resource_

**Comment by hzxuzhonghu** [comment]
As the example shows, we are settinf version as well

**Comment by hzxuzhonghu** [comment]
@howardjohn ptal

```
// name format: <cluster>/<group>/<kind>/<namespace>/<name></section-name>
func (c *Controller) generatePodUID(p *v1.Pod) string {
	return c.clusterID.String() + "//" + "v1/pod/" + p.Namespace + "/" + p.Name
}
```

**Comment by hzxuzhonghu** [comment]
>- If I create a VirtualService, I get a v1alpha3 and v1beta1 that are the same resource

@howardjohn It is not same resource, how to express same name VSes in v1alpha3  and v1beta1.
Though we donot support multi versions.

**Comment by howardjohn** [comment]
they are the same - see
https://blog.howardjohn.info/posts/crd-versioning/#version-conversion. We
have 2 (or more) unique ids for the same resource which is unneeded. Should
just have 1

**Comment by hzxuzhonghu** [comment]
I understand that, so if change all places to remove version, that is inconsistent with k8s url semantics

BTW, `generatePodUID` is not right

**Comment by howardjohn** [comment]
Consistency with k8s url is not a desire - we are already not consistent

File: pilot/pkg/serviceregistry/kube/controller/ambientindex.go [hunk 1]
```diff
@@ -81,7 +81,7 @@ func serviceToAddressInfo(s *workloadapi.Service) *model.AddressInfo {
 
 // name format: <cluster>/<group>/<kind>/<namespace>/<name></section-name>
 func (c *Controller) generatePodUID(p *v1.Pod) string {
-	return c.clusterID.String() + "//" + "v1/pod/" + p.Namespace + "/" + p.Name
+	return c.clusterID.String() + "//" + "Pod/" + p.Namespace + "/" + p.Name
```
Comment (howardjohn) [inline pilot/pkg/serviceregistry/kube/controller/ambientindex.go:84]: nit: prefer `pods` to `Pod`.

**Review by howardjohn — COMMENTED** [review]
cc @kdorosh can you verify this is ok?

**Review by kdorosh — APPROVED** [review]
these changes lgtm

**Comment by hzxuzhonghu** [comment]
ping @howardjohn

**Review by howardjohn — APPROVED** [review]

**Pull request merged**